libs-scala/scala-utils/src/main/scala/com/daml/scalautil/Statement.scala

```diff
@@ -1,9 +1,16 @@
 // Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-package com.daml.http
+package com.daml.scalautil
 
 object Statement {
+
+  /**
+    *  Suppresses `traverser:org.wartremover.warts.NonUnitStatements` warning on the expression level.
+    *
+    * @param evaluateForSideEffectOnly an expression with a side-effect that needs to be evaluated.
+    * @tparam A return type of the expression that gets discarded.
+    */
   @specialized def discard[A](evaluateForSideEffectOnly: A): Unit = {
     val _: A = evaluateForSideEffectOnly
```

Comment:
Why is this line necessary? `evaluateForSideEffectOnly` has already been evaluated at this point; it's not a thunk.

Reply:
I guess it is to avoid the compiler warning. If you do not assign it to `_`, you get:
```
libs-scala/scala-utils/src/main/scala/com/daml/scalautil/Statement.scala:14: warning: parameter value evaluateForSideEffectOnly in method discard is never used
  @specialized def discard[A](evaluateForSideEffectOnly: A): Unit = {
```

@S11001001 do you know if there is any other reason?

In any case the entire function has been "borrowed" from the `NonUnitStatements` wartremover docs:
http://www.wartremover.org/doc/warts.html

Reply:
No, that is the only reason.